### PR TITLE
add enemy name filter

### PR DESCRIPTION
**Pull request by RobbieBendick**
added a textfield to the GUI to allow the user to enter a name, and it'll search for any enemy name with a substring of that value and update the rows accordingly.
For example, "Rob" and "bbi" both match "Robbie"

### CODE REVIEW

**Review by Simon3 — APPROVED** [review]

File: GUI.lua [hunk 1]
```diff
@@ -116,6 +117,14 @@ function ArenaStats:CreateGUI()
     arenaTypeDropdown:SetValue(filters.arenaType)
     asGui.f:AddChild(arenaTypeDropdown)
 
+    local nameFilter = AceGUI:Create("EditBox")
+    nameFilter:SetLabel("Filter By Name")
```
Comment (denishamann) [inline GUI.lua:121]: Do we need to localize this string?

File: GUI.lua [hunk 2]
```diff
@@ -116,6 +117,14 @@ function ArenaStats:CreateGUI()
...
+    nameFilter:SetWidth(150)
+    nameFilter:SetCallback("OnEnterPressed", function(widget, event, text)
```
Comment (denishamann) [inline GUI.lua:123]: How clear is it for the user that he has to press enter after typing his text? If there is not an "ok" or "search" button, I would be in favor of using "OnTextChanged" instead, of course if it's not too ressource intensive.

**Review by denishamann — COMMENTED** [review]
Thanks a lot for your PR! Here are some small comments/questions :) 

Dekos and I are still trying to maintain the addon as best as we can but we are not playing the game anymore. 

Did you test your code in game with the current client ?
 
It would be awesome if you could attach a screenshot of what it looks like.

**Comment by RobbieBendick** [comment]
> It would be awesome if you could attach a screenshot of what it looks like.

of course, here's a gif of what it looks like in action: [here](https://i.imgur.com/PIt6gh9.gif)
as a side note, it works along with the other filters aswell,

File: GUI.lua (hunk 2 above)
Comment (RobbieBendick) [reply GUI.lua:123]: It does have an "okay" button once the value in the textfield has been changed, so in my opinion it is quite obvious that enter being pressed is needed, or to click the okay button.

File: GUI.lua (hunk 1 above)
Comment (RobbieBendick) [reply GUI.lua:121]: Yes, you're definitely correct. Need that string to be localized

**Review by denishamann — APPROVED** [review]

**Comment by denishamann** [comment]
Thanks again for your PR and your example gif. 
I will draft a release asap.

**Pull request merged**